test/runner.js

```diff
@@ -0,0 +1,151 @@
+'use strict'
+
+let fs = require('fs')
+let path = require('path')
+let expect = require('chai').expect
+let rimraf = require('rimraf')
+let Launcher = require('webdriverio/build/lib/launcher')
+let parseXmlString = require('xml2js').parseString
+
+let configFile = './test/fixtures/wdio.conf.js'
+let resultsDir = path.join(__dirname, '../allure-results') 
+
+describe('"before all" hook', () => {
+
+  beforeEach(clean)
+
+  it('should not show up in the results when it is passing', () => {
+
+    return run(['before-all-passing']).then((code) => {
+
+      expect(code, 'wrong exit status code').to.equal(0)
+      return getResultsXML();
+      
+    })
+    .then((results) => {
+
+      expect(results).to.have.lengthOf(1)
+      expect(results[0]['ns2:test-suite']['test-cases']).to.have.lengthOf(1)
+      
+      const hookNames = ['before', 'before all', 'before each', 'after all', 'after each'].map(hookName => '"' + hookName + '" hook');
+      const hookErrors = results[0]['ns2:test-suite']['test-cases'][0]['test-case'].filter(testCase => hookNames.indexOf(testCase.name[0]) > -1)
+      expect(hookErrors, 'there should not be any hooks').to.be.empty()
+
+    })
+
+  })
+
+  it('should marked broken when failing', () => {
+
+    return run(['before-all-failing']).then((code) => {
+
+      expect(code, 'wrong exit status code').to.equal(1)
```

Comment:
I think it's better to check that it has a non-zero value, because some libs / commands can exit with code greater then 1 and it is still error.